pkgs/dart_mcp/lib/src/api/sampling.dart

```diff
@@ -4,267 +4,201 @@
+    MetaWithProgressToken? meta,
+  }) => CreateMessageRequest.fromMap({
+    'messages': messages,
+    if (modelPreferences != null) 'modelPreferences': modelPreferences,
+    if (systemPrompt != null) 'systemPrompt': systemPrompt,
+    if (includeContext != null) 'includeContext': includeContext.name,
+    if (temperature != null) 'temperature': temperature,
+    'maxTokens': maxTokens,
+    if (stopSequences != null) 'stopSequences': stopSequences,
+    if (metadata != null) 'metadata': metadata,
+    if (meta != null) '_meta': meta,
+  });
+
+  /// The messages to send to the LLM.
+  List<SamplingMessage> get messages =>
+      (_value['messages'] as List).cast<SamplingMessage>();
+
+  /// The server's preferences for which model to select. The client MAY
+  /// ignore these preferences.
+  ModelPreferences? get modelPreferences =>
+      _value['modelPreferences'] as ModelPreferences?;
+
+  /// An optional system prompt the server wants to use for sampling. The
+  /// client MAY modify or omit this prompt.
+  String? get systemPrompt => _value['systemPrompt'] as String?;
+
+  /// A request to include context from one or more MCP servers (including
+  /// the caller), to be attached to the prompt. The client MAY ignore this
+  /// request.
+  IncludeContext? get includeContext {
+    final includeContext = _value['includeContext'] as String?;
+    if (includeContext == null) return null;
+    return IncludeContext.values.firstWhere((c) => c.name == includeContext);
+  }
 
-// /**
-//  * Describes a message issued to or received from an LLM API.
-//  */
-// export interface SamplingMessage {
-//   role: Role;
-//   content: TextContent | ImageContent;
-// }
+  /// The temperature to use for sampling
+  double? get temperature => _value['temperature'] as double?;
 
-// /**
-//  * Base for objects that include optional annotations for the client. The
-//  * client can use annotations to inform how objects are used or displayed
-//  */
-// export interface Annotated {
-//   annotations?: {
-//     /**
-//      * Describes who the intended customer of this object or data is.
-//      *
-//      * It can include multiple entries to indicate content useful for
-//      * multiple audiences (e.g., `["user", "assistant"]`).
-//      */
-//     audience?: Role[];
+  /// The maximum number of tokens to sample, as requested by the server.
+  ///
+  /// The client MAY choose to sample fewer tokens than requested.
+  int get maxTokens => _value['maxTokens'] as int;
 
-//     /**
-//      * Describes how important this data is for operating the server.
-//      *
-//      * A value of 1 means "most important," and indicates that the data is
-//      * effectively required, while 0 means "least important," and indicates
-//      * that the data is entirely optional.
-//      *
-//      * @TJS-type number
-//      * @minimum 0
-//      * @maximum 1
-//      */
-//     priority?: number;
-//   }
-// }
+  List<String>? get stopSequences =>
```

Comment:
dart doc?

Reply:
I honestly don't know what this is, its not documented [here](https://spec.modelcontextprotocol.io/specification/2025-03-26/client/sampling/) or in the [schema](https://github.com/modelcontextprotocol/specification/blob/main/schema/2024-11-05/schema.json#L373).

I can add a TODO or just a note that its undocumented.